### DISCUSSION TITLE
Artillery changing case of header name to lower case which is making api call to fail.

### DISCUSSION THREAD

**Discussion by BhavnaSoni1991** [artilleryio/artillery#3284] [Artillery]
I am trying to write performance test for an api which requires custom header. When I am passing this header name under headers in yml file artillery is changing the case of this header name to lower case which is failing the test.

I am not setting this header name using any javascript method it is directly provided in yml file. Can someone please suggest any solution for this?
Thanks in advance!!

**Comment by BhavnaSoni1991** [comment]
@hassy , @bernardobridge Can anyone please comment if the fix for this issue is rolled back in latest releases as I am facing this issue in my test where the header name is getting changed to lower case and the tests are failing because of that?

**Comment by bernardobridge** [comment]
Hello @BhavnaSoni1991,

There isn't an easy fix for this, because technically it is not a bug. HTTP header field names are case-insensitive as per the [HTTP spec RFC](https://www.rfc-editor.org/rfc/rfc7230#section-3.2):

<img width="622" alt="Screenshot 2024-07-29 at 10 28 36" src="https://github.com/user-attachments/assets/6ef6eb9b-849c-4d76-afc3-c60c051a6017">

The underlying library we use ([`got`](https://www.npmjs.com/package/got)) follows this spec, and lower cases the headers. Overriding this behaviour is not straightforward, and it's not something we can do as a quick fix.

The only thing I can suggest to you is to use a [custom function](https://www.artillery.io/docs/reference/engines/http#function-steps-in-a-flow) to make the HTTP call yourself, using whatever HTTP library you want that maybe doesn't do this. Or, ideally, the application is not going against the spec by relying on headers being case sensitive (which I understand can be hard in certain legacy applications).

**Reply by BhavnaSoni1991** [reply]
Thanks for your response @bernardobridge!!
